linera-service/tests/end_to_end_tests.rs

```diff
@@ -562,6 +594,227 @@ async fn test_wasm_end_to_end_same_wallet_fungible(config: impl LineraNetConfig)
+        (account_owner1, Amount::from_tokens(5)),
+        (account_owner2, Amount::from_tokens(2)),
+    ]);
+    let state = InitialState { accounts };
+    // Setting up the application and verifying
+    let (contract, service) = client1.build_example("native-fungible").await.unwrap();
+    let params = fungible::Parameters::new("FUN");
+    let application_id = client1
+        .publish_and_create::<FungibleTokenAbi>(contract, service, &params, &state, &[], None)
+        .await
+        .unwrap();
+
+    let mut node_service1 = client1.run_node_service(8080).await.unwrap();
+    let mut node_service2 = client2.run_node_service(8081).await.unwrap();
+
+    let app1 = NativeFungibleApp(FungibleApp(
+        node_service1
+            .make_application(&chain1, &application_id)
+            .await
+            .unwrap(),
+    ));
+    app1.assert_balances([
+        (account_owner1, Amount::from_tokens(5)),
+        (account_owner2, Amount::from_tokens(2)),
+    ])
+    .await;
+
+    // Transferring
+    app1.transfer(
+        &account_owner1,
+        Amount::ONE,
+        fungible::Account {
+            chain_id: chain2,
+            owner: account_owner2,
+        },
+    )
+    .await;
+
+    // Checking the final values on chain1 and chain2.
+    app1.assert_balances([
+        (account_owner1, Amount::from_tokens(4)),
+        (account_owner2, Amount::from_tokens(2)),
+    ])
+    .await;
+
+    node_service2
+        .request_application(&chain2, &application_id)
+        .await
+        .unwrap();
+    // Fungible didn't exist on chain2 initially but now it does and we can talk to it.
+    let app2 = NativeFungibleApp(FungibleApp(
+        node_service2
+            .make_application(&chain2, &application_id)
+            .await
+            .unwrap(),
+    ));
+
+    app2.assert_balances(BTreeMap::from([
+        (account_owner1, Amount::ZERO),
+        (account_owner2, Amount::ONE),
+    ]))
+    .await;
+
+    // Claiming more money from chain1 to chain2.
+    app2.claim(
+        fungible::Account {
+            chain_id: chain1,
+            owner: account_owner2,
+        },
+        fungible::Account {
+            chain_id: chain2,
+            owner: account_owner2,
+        },
+        Amount::from_tokens(2),
+    )
+    .await;
+
+    // Make sure that the cross-chain communication happens fast enough.
+    node_service1.process_inbox(&chain1).await.unwrap();
+    node_service2.process_inbox(&chain2).await.unwrap();
+
+    // Checking the final value
+    app1.assert_balances([
+        (account_owner1, Amount::from_tokens(4)),
+        (account_owner2, Amount::ZERO),
+    ])
+    .await;
+    app2.assert_balances([
+        (account_owner1, Amount::ZERO),
+        (account_owner2, Amount::from_tokens(3)),
+    ])
+    .await;
+
+    node_service1.ensure_is_running().unwrap();
+    node_service2.ensure_is_running().unwrap();
+
+    net.ensure_is_running().await.unwrap();
+    net.terminate().await.unwrap();
```

Comment:
It's not identical because we build the `native-fungible` example instead, use the `NativeFungibleApp` instead, the ticker will be fixed and we also have to `request_application` before calling `make_application` for chain2, because on the system API calls we won't register the application on chain2 when we transfer to it, like it happens for `fungible`.
We could try to parameterize all this, but I wouldn't be surprised if these tests eventually diverge, so didn't seem worth the effort. It would also be a lot of parameterization

Reply:
On the other hand I think we want to keep them as compatible as possible. Not sure in how many places they would be different?

Reply:
I see. Let me see if I can unify it